documentation/decisions/01 - Using a Compound Component Pattern for Autocomplete.md

```diff
@@ -0,0 +1,28 @@
+# Using a compound component patter to build Autocomplete
+
+## Date
+
+October 3, 2019
+
+## Contributors
+
+- Andrew Musgrave
+- Chloe Rice
+- Dan Rosenthal
+- Daniel Leroux
+- Dominic McPhee
+- Sara Hill
+
+## Summary
+
+There is a need for Polaris to provide smaller more reusable components so that Polaris can be used by everyone without needing to fork component. The re-development of the Autocomplete component provided an opportunity for exploration.
+
+## Solution
+
+Following an (https://github.com/Shopify/polaris-react/issues/1794)[exploration] we decided to build the Autocomplete using a compound component pattern. We explored various ways to do this:
+
+1. Having the parent render its children via methods shared via Context: We opted against this because it defied the purpose of having smaller components and cause the component to renderer every time an option as added the Autocompletes results. For example, 100 results cause for 100 calls to the parent.
```

Comment:
```suggestion
1. Having the child delegate rendering to its parent via methods shared through context

We opted against this for a couple of reasons:
- it defeated the purpose of having smaller components
- it causes the component to re-render every time an option is added (one call to the parent render method for each option)
```

Reply:
> - it defeated the purpose of having smaller components

This needs more detail. Why did it defeat the purpose of having smaller components?